spec/buffer-modifying-provider-spec.coffee

```diff
@@ -0,0 +1,77 @@
+describe 'buffer modifying linters', ->
+  getModuleMain = -> atom.packages.getActivePackage('linter').mainModule.instance
+  beforeEach ->
+    waitsForPromise ->
+      atom.packages.activatePackage('linter')
+    waitsForPromise ->
+      atom.workspace.open('test.txt')
+  it 'are triggered before other linters', ->
+    linter = getModuleMain()
+    last = null
+    bufferModifying =
+      grammarScopes: ['*']
+      scope: 'file'
+      lintOnFly: false
+      modifiesBuffer: true
+      lint: ->
+        last = 'bufferModifying'
+        return []
+    normalLinter =
+      grammarScopes: ['*']
+      scope: 'file'
+      lintOnFly: false
+      modifiesBuffer: false
+      lint: ->
+        last = 'normal'
+        return []
+    linter.addLinter(bufferModifying)
+    linter.addLinter(normalLinter)
+    waitsForPromise ->
+      linter.getActiveEditorLinter().lint(false).then ->
+        expect(last).toBe('normal')
+  it 'run in sequence', ->
```

Comment:
we are actually describing `buffer modifying linters` above, so we don't have to write that whole message, when we write `run in sequence` it becomes `buffer modifying linters run in sequence`